File: sqlparser/src/sql/NormalizationOptionBuilder.java
//@
package cdata.sql;
import java.util.Enumeration;
import java.util.Hashtable;
//@
/*#
using RSSBus.core;
using RSSBus;
namespace CData.Sql {
#*/

public class NormalizationOptionBuilder {
  private Hashtable<NormalizationOptions, Object> _options = new Hashtable<NormalizationOptions, Object>();

  public NormalizationOptionBuilder config(NormalizationOptions option, Object val) {
    this.setOption(option, val);
    return this;
  }

  public String getConfigAsString(NormalizationOptions configName) {
    return getConfigAsString(configName, "");
  }

  public String getConfigAsString(NormalizationOptions configName, String nullValue) {
    Object objValue = getOption(configName);
    if (objValue instanceof String) {
      return (String)objValue;
    }
    return nullValue;
  }

  public String[] getConfigAsStringArray(NormalizationOptions configName) {
    return this.getConfigAsStringArray(configName, new String[0]);
  }

  public String[] getConfigAsStringArray(NormalizationOptions configName, String[] nullValue) {
    Object objValue = getOption(configName);
    if (objValue instanceof String[]) {
      return ( String[])objValue;
    }
    return nullValue;
  }

  public boolean getConfigAsBoolean(NormalizationOptions configName) {
    return this.getConfigAsBoolean(configName, false);
  }

  public boolean getConfigAsBoolean(NormalizationOptions configName, boolean nullValue) {
    Object objValue = getOption(configName);
    if (objValue instanceof Boolean) {
      return (Boolean)objValue;
    }
    return nullValue;
  }

  public int getConfigAsInteger(NormalizationOptions configName) {
    return this.getConfigAsInteger(configName, -1);
  }

  public int getConfigAsInteger(NormalizationOptions configName, int nullValue) {
    Object objValue = getOption(configName);
    if (objValue instanceof Integer) {
      return (Integer)objValue;
    }
    return nullValue;
  }

  public NormalizationOption build() {
    return new NormalizationOption(this);
  }

  public void clear() {
    this._options.clear();
  }

  public void copy(NormalizationOptionBuilder obj) {
    Enumeration e = obj._options.keys();
    while (e.hasMoreElements()) {
      NormalizationOptions key = (NormalizationOptions)e.nextElement();
      this._options.put(key, obj._options.get(key));
    }
  }

  private void setOption(NormalizationOptions configName, Object configValue) {
    if (!this._options.containsKey(configName)) {
      this._options.put(configName, configValue);
    }
  }

  private Object getOption(NormalizationOptions configName) {
    if (this._options.containsKey(configName)) {
      return this._options.get(configName);
    }
    return null;
  }
}

File: sqlparser/src/sql/csgen/NormalizationOptionBuilder.cs
using System;
using System.Collections;
using System.IO;
using System.Threading;
using System.Net;
using System.Net.Sockets;
using System.Text;
using RSSBus.core.j2cs;



using RSSBus.core;
using RSSBus;
namespace CData.Sql {


public class NormalizationOptionBuilder {
  private JavaHashtable<NormalizationOptions, Object> _options = new JavaHashtable<NormalizationOptions, Object>();

  public NormalizationOptionBuilder Config(NormalizationOptions option, Object val) {
    this.SetOption(option, val);
    return this;
  }

  public string GetConfigAsString(NormalizationOptions configName) {
    return GetConfigAsString(configName, "");
  }

  public string GetConfigAsString(NormalizationOptions configName, string nullValue) {
    Object objValue = GetOption(configName);
    if (objValue is string) {
      return (string)objValue;
    }
    return nullValue;
  }

  public string[] GetConfigAsStringArray(NormalizationOptions configName) {
    return this.GetConfigAsStringArray(configName, new string[0]);
  }

  public string[] GetConfigAsStringArray(NormalizationOptions configName, string[] nullValue) {
    Object objValue = GetOption(configName);
    if (objValue is string[]) {
      return ( string[])objValue;
    }
    return nullValue;
  }

  public bool GetConfigAsBoolean(NormalizationOptions configName) {
    return this.GetConfigAsBoolean(configName, false);
  }

  public bool GetConfigAsBoolean(NormalizationOptions configName, bool nullValue) {
    Object objValue = GetOption(configName);
    if (objValue is Boolean) {
      return (Boolean)objValue;
    }
    return nullValue;
  }

  public int GetConfigAsInteger(NormalizationOptions configName) {
    return this.GetConfigAsInteger(configName, -1);
  }

  public int GetConfigAsInteger(NormalizationOptions configName, int nullValue) {
    Object objValue = GetOption(configName);
    if (objValue is int) {
      return (int)objValue;
    }
    return nullValue;
  }

  public NormalizationOption Build() {
    return new NormalizationOption(this);
  }

  public void Clear() {
    this._options.Clear();
  }

  public void Copy(NormalizationOptionBuilder obj) {
    JavaEnumeration e = obj._options.Keys();
    while (e.HasMoreElements()) {
      NormalizationOptions key = (NormalizationOptions)e.NextElement();
      this._options.Put(key, obj._options.Get(key));
    }
  }

  private void SetOption(NormalizationOptions configName, Object configValue) {
    if (!this._options.ContainsKey(configName)) {
      this._options.Put(configName, configValue);
    }
  }

  private Object GetOption(NormalizationOptions configName) {
    if (this._options.ContainsKey(configName)) {
      return this._options.Get(configName);
    }
    return null;
  }
}
}

